### PR TITLE
Setup MSW 

### LINKED ISSUE #57: Setup MSW and Refactor API Mocking Tests
**Description:**

Our current testing infrastructure lacks efficient API mocking, hindering our ability to conduct reliable and controlled tests. To address this, we aim to integrate Mock Service Worker (MSW) into our testing environment and refactor existing tests for improved API mocking.

**Proposed Solution:**
- Integration of MSW: Implement MSW to facilitate API mocking in our testing suite.

- Refactor Existing Tests: Review and refactor existing tests to align them with the new MSW setup and take advantage of its features for better control and reliability.

**Steps to Reproduce (if applicable):**

- Attempt to run existing tests related to API interactions.

- Observe the absence of effective API mocking.

- Implement MSW and refactor tests accordingly.

- Re-run tests to ensure successful API mocking and reliable test outcomes.

**Expected Outcome:**

- Successful integration of MSW.

- Refactored tests reflecting the new API mocking setup.

- Existing tests passing with the updated configuration.

**Additional Information:**

- Dependencies/Requirements:

- Ensure documentation is updated to reflect the new testing setup.

- Collaborate with team members to address potential conflicts with existing codebase.

- Test the changes in different environments to ensure compatibility.

### PR DESCRIPTION
**Pull request by vinit717**
## Issue Ticket Number

-   Closes #57 

## Description

Setup MSW and handler with test

**Breaking Changes**

-   [ ] Yes
-   [x] No

_If your feature introduces breaking changes or if something is missing, please mention the related issue tickets._

**Development Tested?**

-   [x] Yes
-   [ ] No

_Confirm whether the changes have been tested locally during development._

**Tested in Staging?**

-   [ ] Yes
-   [x] No

_Indicate whether the changes have been tested in the staging environment._

**Under Feature Flag**

-   [x] Yes
-   [ ] No

_Specify if the changes are currently under a feature flag._

**Database Changes**

-   [ ] Yes
-   [x] No

_Indicate whether the changes include modifications to the database._

### Screenshots

N/A.

## Test Coverage

N/A

## Additional Notes

Include any additional notes, considerations, or explanations that might be helpful for reviewers.

### CODE REVIEW

File: __mocks__/handlers/user.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { rest } from 'msw';
+
+import { TINY_API_URL } from '@/constants/url';
+
+import userData from '../db/user';
+
+const userHandler = [
```
Comment (sahsisunny) [inline __mocks__/handlers/user.ts:7]:
```suggestion
const getSelfUserHandler = [
```

File: __tests__/utils/fetchUrls.test.ts [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
+
+        const result = await fetchUrls(userOne);
+
         expect(result).toBeNull();
     });
-    it('handles null response', async () => {
-        (axios.get as jest.Mock).mockResolvedValue(null);
-        const result = await fetchUrls(user);
+
+    it('should handle errors and return null', async () => {
```
Comment (sahsisunny) [inline __tests__/utils/fetchUrls.test.ts:23]: Test titles should be different

Comment (vinit717) [reply __tests__/utils/fetchUrls.test.ts]: tests are removed

File: __mocks__/handlers/user.ts (hunk 1 above)
Comment (vinit717) [reply __mocks__/handlers/user.ts]: done

File: __mocks__/handlers/url.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { rest } from 'msw';
+
+import { TINY_API_URL } from '@/constants/url';
+
+import urls from '../db/urls';
+
+const urlHandler = [
```
[the remark this answers is not in the document]
Comment (vinit717) [reply __mocks__/handlers/url.ts]: done

**Review by prakashchoudhary07 — APPROVED** [review]

File: __mocks__/handlers/user.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import userData from '../db/user';
+
+const getSelfUserHandler = [
+    rest.get(`${TINY_API_URL}/users/self`, (_, res, ctx) => {
+        return res(ctx.status(200), ctx.json(userData));
+    }),
+];
+
+const failedUserHandler = [
```
Comment (sahsisunny) [inline __mocks__/handlers/user.ts:13]: Also for this

**Review by sahsisunny — DISMISSED** [review]

File: __mocks__/handlers/url.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        ctx.status(404),
+        ctx.json({
+            statusCode: 404,
+            error: 'Not Found',
+            message: 'urls not found',
+        })
+    );
+});
+
+const failedUrls = rest.get(`${TINY_API_URL}/user/1/urls`, (_, res) => {
```
[the remark this answers is not in the document]
Comment (vinit717) [reply __mocks__/handlers/url.ts]: done

File: __mocks__/handlers/user.ts (hunk 4 above)
Comment (vinit717) [reply __mocks__/handlers/user.ts]: done

**Review by sahsisunny — APPROVED** [review]

File: __mocks__/handlers/user.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+const getSelfUserHandler = [
+    rest.get(`${TINY_API_URL}/users/self`, (_, res, ctx) => {
+        return res(ctx.status(200), ctx.json(userData));
+    }),
+];
+
+const unauthorizedUserHandler = [
+    rest.get(`${TINY_API_URL}/users/self`, (req, res, ctx) => {
+        return res(ctx.status(401));
```
[the remark this answers is not in the document]
Comment (vinit717) [reply __mocks__/handlers/user.ts]: added thanks for pointing

**Review by shubhamsigdar1 — APPROVED** [review]
LGTM

**Review by satyam73 — APPROVED** [review]
LGTM

**Pull request merged**